server/src/main/java/org/opensearch/search/aggregations/bucket/terms/StreamStringTermsAggregator.java

```diff
@@ -91,6 +93,11 @@ public InternalAggregation buildEmptyAggregation() {
 
     @Override
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub) throws IOException {
+        if (this.madeLeafOnce) {
+            throw new IllegalStateException("pardon you might already collected " + docCounts);
```

Comment:
what's the proper exception here?

Reply:
IllegalStateException would ultimately translate back to a 500 internal service error. I think that's right here because if this invariant is broken it's a bug and not any sort of user error. Does that sound right?